src/file-loader.ts

```diff
@@ -182,17 +184,20 @@ export class FileDataLoader {
         throw new Error(`Unsupported data type: ${this.config.type}`);
       }
     }
+  }
 
-    function mapDocumentsIntoInsertPayload(
-      splittedDocuments: Document[],
-      metadataMapper?: (metadata: any, index: number) => Record<string, any>
-    ) {
-      return splittedDocuments.map((document, index) => ({
-        data: document.pageContent,
-        id: nanoid(),
-        ...(metadataMapper ? { metadata: metadataMapper(document.metadata, index) } : {}),
-      }));
-    }
+  private mapDocumentsIntoInsertPayload(
+    splittedDocuments: Document[],
+    metadataMapper?: (metadata: any, index: number) => Record<string, any>
+  ) {
+    return splittedDocuments.map((document, index) => ({
+      data: document.pageContent,
+      id: nanoid(),
+      metadata: {
+        ...(metadataMapper ? metadataMapper(document.metadata, index) : {}),
+        ...this.config.options?.metadata,
```

Comment:
this was the biggest change here, 

I'm destructuring the current metadata, to accept the provided metadata by the user, as we can do on raw texts